plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -606,6 +609,11 @@ private ConnectorTableMetadata doGetTableMetadata(ConnectorSession session, Sche
 
         Optional<String> comment = Optional.ofNullable(table.getParameters().get(TABLE_COMMENT));
 
+        String autoPurgeProperty = table.getParameters().get(AUTO_PURGE_KEY);
+        if (parseBoolean(autoPurgeProperty)) {
+            properties.put(AUTO_PURGE, true);
```

Comment:
This will make the property stored and available to metastore.

What if it's _us_ who delete the data? 
if user can set `auto_purge`, they would expect us to obey it.

we can delete data eg in glue metastore flow, so next question -- does this property make sense when metastore is Glue?

Reply:
@findepi from my understanding of Glue, this property does not make sense when the metastore is Glue.

The [DeleteTable](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-catalog-tables.html#aws-glue-api-catalog-tables-DeleteTable) request states the data for the table will be deleted asynchronously which means the `auto_purge` property would have no effect if it was set to `false` i.e. data will always be purged.